snuba/datasets/storages/generic_metrics.py

```diff
@@ -181,3 +192,49 @@ def produce_policy_creator() -> DeadLetterQueuePolicy:
         pre_filter=KafkaHeaderSelectFilter("metric_type", InputType.DISTRIBUTION.value),
     ),
 )
+
+
+file = open(
+    "./snuba/datasets/configuration/generic_metrics/storage_distributions_raw.yaml"
+)
+conf_yml = safe_load(file)
+assert isinstance(conf_yml, dict)
+conf = dataclass_from_dict(StorageConfig, conf_yml)
+assert isinstance(conf, StorageConfig)
```

Comment:
What is the process to instantiate a storage from config ?
Do we expect the storages to be still created statically at module initialization ? And is the factory going to still have a role ?

Reply:
What do you mean by "process"?

I do expect to have Storages built on module initialization, since that is how they are currently initialized. We are trying to avoid changing any existing logic we don't have too. The factory will not have a role in configuration storages, but the `get_storage` function will still be used.